File: BancoCoreSOAP/src/main/java/com/example/BancoCoreSoap/repository/ContaRepository.java
package com.example.BancoCoreSoap.repository;

import com.example.BancoCoreSoap.domain.Conta;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import org.springframework.data.jpa.repository.Lock;
import jakarta.persistence.LockModeType;

@Repository
public interface ContaRepository extends JpaRepository<Conta, String>{
	
	@Lock(LockModeType.PESSIMISTIC_WRITE)
	Optional<Conta> findByNumeroConta(String numeroConta);
}

File: BancoRestApi/src/main/kotlin/org/example/bancorestapi/repository/ContaRepository.kt
package org.example.bancorestapi.repository

import jakarta.persistence.LockModeType
import org.example.bancorestapi.domain.ChavePix
import org.example.bancorestapi.domain.Conta
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Lock
import org.springframework.stereotype.Repository

@Repository
interface ContaRepository : JpaRepository<Conta, String> {

    @Lock(LockModeType.PESSIMISTIC_WRITE)
    fun findByNumeroConta(numeroConta: String): Conta?

    @Lock(LockModeType.PESSIMISTIC_WRITE)
    fun findById(id: Long): Conta?
}